packages/shell-api/src/decorators.ts

```diff
@@ -142,6 +155,8 @@ export interface TypeSignature {
   deprecated?: boolean;
   returnType?: string | TypeSignature;
   attributes?: { [key: string]: TypeSignature };
+  isDirectShellCommand?: boolean;
```

Comment:
Is this used only for autocompletion or also to set up the command in the various shells? Would be great if we have to have this for it to be the source of truth about that. Hope i've understood what this means.

I think a comment here could help, the concept of "direct command" may not be immediate without context.

Reply:
> Is this used only for autocompletion or also to set up the command in the various shells? Would be great if we have to have this for it to be the source of truth about that. Hope i've understood what this means.

The property on the function instance itself is what the shell evaluator uses to determine whether to allow bash-style calls to it, the property on the signature is what the autocompleter uses (like all other properties).

In either case, the source of truth is the `@directShellCommand` decorator.

> I think a comment here could help, the concept of "direct command" may not be immediate without context.

Good idea, done! :+1:

Reply:
Actually now that you called it that way .. is "bash-style" command a name that's even more clear? Of course is fine as is.

Reply:
I mean, it’s not really bash-specific, it’s like that in most OS shells (even Windows cmd.exe) :)